types/field.go

```diff
@@ -37,3 +41,47 @@ func (l FieldsDefinition) Names() []string {
 	}
 	return names
 }
+
+// SelectedField is the public representation of a field selection during a graphql query
+type SelectedField struct {
+	Name       string
+	Alias      string
+	Directives DirectiveList
+	Args       map[string]interface{}
+	Fields     SelectedFieldList
+}
+
+type SelectedFieldList []*SelectedField
+
+func (sf SelectedFieldList) Names() (res []string) {
+	for _, f := range sf {
+		res = append(res, f.Name)
+	}
+	return
+}
+
+// Lookup searches the field tree for the field defined by the input path.
+// The path supports aliases so that an aliased field can be pointed to with the following syntax: `alias:fieldName`.
+func (f *SelectedField) Lookup(path ...string) *SelectedField {
```

Comment:
i think we should have a custom struct type for the path, instead of parsing the strings